modular_skyrat/modules/xenoarch/code/modules/research/xenoarch/strange_rock.dm

```diff
@@ -0,0 +1,212 @@
+/obj/item/xenoarch/strange_rock
+	name = "strange rock"
+	desc = "A mysterious, strange rock that has the potential to have a wonderful item. Also possible for it to have our disposed garbage."
+	icon_state = "rock"
+
+	var/max_depth //the max depth
+	var/safe_depth //the depth away from the max depth
+	var/item_depth //the depth between the max and the safe, containing the item
+	var/dug_depth = 0 //the depth that has been currently dug
+
+	var/hidden_item //the item that is held
+
+	var/measured = FALSE //if the item has been tape-measured
+	var/scanned = FALSE //if the item has been scanned
+
+	var/scan_state = "rock_Strange"
+
+/obj/item/xenoarch/strange_rock/Initialize()
+	. = ..()
+	create_item()
+	create_depth()
+
+/obj/item/xenoarch/strange_rock/examine(mob/user)
+	. = ..()
+	. += span_notice("[scanned ? "This item has been scanned. Max Depth: [max_depth] cm. Safe Depth: [safe_depth] cm." : "This item has not been scanned."]")
+	. += span_notice("[measured ? "This item has been measured. Dug Depth: [dug_depth]." : "This item has not been measured."]")
+	if(measured && dug_depth > item_depth)
+		. += span_warning("The rock is crumbling, even just brushing it will destroy it!")
+
+/obj/item/xenoarch/strange_rock/proc/create_item()
+	var/choose_tier = rand(1,100)
+	switch(choose_tier)
+		if(1 to 70)
+			hidden_item = pickweight(GLOB.tier1_reward)
+		if(71 to 97)
+			hidden_item = pickweight(GLOB.tier2_reward)
+		if(98 to 100)
+			hidden_item = pickweight(GLOB.tier3_reward)
+
+/obj/item/xenoarch/strange_rock/proc/create_depth()
+	max_depth = rand(21, 100)
+	safe_depth = rand(1, 10)
+	item_depth = rand((max_depth - safe_depth), max_depth)
+	dug_depth = rand(0, 10)
+
+/obj/item/xenoarch/strange_rock/proc/get_measured() //returns true if successful measure
+	if(measured)
+		return FALSE
+	measured = TRUE
+	return TRUE
+
+/obj/item/xenoarch/strange_rock/proc/get_scanned() //returns true if successful scan
+	if(scanned)
+		return FALSE
+	scanned = TRUE
+	return TRUE
+
+/obj/item/xenoarch/strange_rock/proc/try_dig(var/dig_amount) //1: no dig amount 2: above dig depth 3: successful dig
+	if(!dig_amount)
+		return 1
+	dug_depth += dig_amount
+	if(dug_depth > item_depth)
+		qdel(src)
+		return 2
+	return 3
+
+/obj/item/xenoarch/strange_rock/proc/try_uncover() //1: brush too above item 2: brush successful 3: brush unsuccesful but fine
+	if(dug_depth > item_depth)
+		qdel(src)
+		return 1
+	if(dug_depth == item_depth)
+		new hidden_item(get_turf(src))
+		qdel(src)
+		return 2
+	return 3
```

Comment:
1, 2 and 3 should be turned into defines to help with readability and remove the need for those comments.